### ISSUE TITLE
DontRemoveLineContinuationInStringInterpolation3 test asserts

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#6341]
As part of the test cleanup, we're disabling tests that are flaky based on Azure runs that we're doing, this test is one of them.

The test is not yet disabled, when it is, the associated commit will be linked.

__Dump:__

`\\mlangfs1\public\davkean\6341.dmp`

__Code:__
``` VB

' Another } may follow the close brace of an interpolation if the interpolation lacked a format clause.
' This is because the normal escaping rules only apply when parsing the format string.
Debug.Assert(Not CanGet(1) OrElse Peek(offset + 1) <> c OrElse IsRightCurlyBracket(c), "Escape sequence not detected.")
```
__Stack:__

```
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Scanner.ScanInterpolatedStringPunctuation() Line 34	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Scanner.GetScannerToken(Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.ScannerState state) Line 622	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Scanner.GetCurrentToken() Line 228	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.get_CurrentToken() Line 5939	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseInterpolatedStringExpression() Line 40	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseTerm(Boolean BailIfFirstTokenRejected, Boolean RedimOrNewParent) Line 411	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseExpressionCore(Microsoft.CodeAnalysis.VisualBasic.Syntax.OperatorPrecedence pendingPrecedence, Boolean bailIfFirstTokenRejected) Line 109	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseFieldOrPropertyAsClauseAndInitializer(Boolean isProperty, Boolean allowAsNewWith, Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.AsClauseSyntax optionalAsClause, Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.EqualsValueSyntax optionalInitializer) Line 2311	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseVariableDeclaration(Boolean allowAsNewWith) Line 2193	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseVarDeclStatement(Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.SyntaxList(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.AttributeListSyntax) attributes, Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.SyntaxList(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.KeywordSyntax) modifiers) Line 2111	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseSpecifierDeclaration(Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.SyntaxList(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.AttributeListSyntax) attributes, Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.SyntaxList(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.KeywordSyntax) modifiers) Line 1331	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseSpecifierDeclaration(Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.SyntaxList(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.AttributeListSyntax) attributes) Line 1294	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseSpecifierDeclaration() Line 1289	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseDeclarationStatementInternal() Line 723	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseDeclarationStatement() Line 644	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.DeclarationContext.Parse() Line 19	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseCompilationUnitCore() Line 488	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseWithStackGuard(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.CompilationUnitSyntax)(System.Func(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.CompilationUnitSyntax) parseFunc, System.Func(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.CompilationUnitSyntax) defaultFunc) Line 512	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.Parser.ParseCompilationUnit() Line 437	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxTree.ParseText(Microsoft.CodeAnalysis.Text.SourceText text, Boolean isMyTemplate, Microsoft.CodeAnalysis.VisualBasic.VisualBasicParseOptions options, String path, System.Threading.CancellationToken cancellationToken) Line 230	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxTree.ParseText(Microsoft.CodeAnalysis.Text.SourceText text, Microsoft.CodeAnalysis.VisualBasic.VisualBasicParseOptions options, String path, System.Threading.CancellationToken cancellationToken) Line 209	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.SyntaxFactory.ParseSyntaxTree(Microsoft.CodeAnalysis.Text.SourceText text, Microsoft.CodeAnalysis.ParseOptions options, String path, System.Threading.CancellationToken cancellationToken) Line 57	Basic
 	Microsoft.CodeAnalysis.VisualBasic.Workspaces.dll!Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxTreeFactoryServiceFactory.VisualBasicSyntaxTreeFactoryService.ParseSyntaxTree(String fileName, Microsoft.CodeAnalysis.ParseOptions options, Microsoft.CodeAnalysis.Text.SourceText text, System.Threading.CancellationToken cancellationToken) Line 36	Basic
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.DocumentState.FullyParseTreeAsync(Roslyn.Utilities.ValueSource<Microsoft.CodeAnalysis.TextAndVersion> newTextSource, Microsoft.CodeAnalysis.ProjectId cacheKey, string filePath, Microsoft.CodeAnalysis.ParseOptions options, Microsoft.CodeAnalysis.Host.HostLanguageServices languageServices, Microsoft.CodeAnalysis.SolutionServices solutionServices, Microsoft.CodeAnalysis.PreservationMode mode, System.Threading.CancellationToken cancellationToken) Line 125	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.DocumentState.CreateLazyFullyParsedTree.AnonymousMethod__0(System.Threading.CancellationToken c) Line 104	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Roslyn.Utilities.AsyncLazy<Microsoft.CodeAnalysis.TreeAndVersion>.StartAsynchronousComputation(Roslyn.Utilities.AsyncLazy<Microsoft.CodeAnalysis.TreeAndVersion>.AsynchronousComputationToStart computationToStart, Roslyn.Utilities.AsyncLazy<Microsoft.CodeAnalysis.TreeAndVersion>.Request requestToCompleteSynchronously, System.Threading.CancellationToken callerCancellationToken) Line 378	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Roslyn.Utilities.AsyncLazy<Microsoft.CodeAnalysis.TreeAndVersion>.GetValueAsync(System.Threading.CancellationToken cancellationToken) Line 331	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.DocumentState.GetSyntaxTreeAsync(System.Threading.CancellationToken cancellationToken) Line 536	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.Document.GetSyntaxTreeAsync(System.Threading.CancellationToken cancellationToken) Line 200	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.Document.GetSyntaxRootAsync(System.Threading.CancellationToken cancellationToken) Line 225	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.CodeCleanup.AbstractCodeCleanerService.CleanupAsync(Microsoft.CodeAnalysis.Document document, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Text.TextSpan> spans, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.CodeCleanup.Providers.ICodeCleanupProvider> providers, System.Threading.CancellationToken cancellationToken) Line 28	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.CodeCleanup.CodeCleaner.CleanupAsync(Microsoft.CodeAnalysis.Document document, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Text.TextSpan> spans, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.CodeCleanup.Providers.ICodeCleanupProvider> providers, System.Threading.CancellationToken cancellationToken) Line 80	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.CodeCleanup.CodeCleaner.CleanupAsync(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.TextSpan span, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.CodeCleanup.Providers.ICodeCleanupProvider> providers, System.Threading.CancellationToken cancellationToken) Line 70	C#
 	Roslyn.Services.UnitTests.dll!Microsoft.CodeAnalysis.UnitTests.CodeCleanup.RemoveUnnecessaryLineContinuationTests.Verify(string codeWithMarker, string expectedResult, Microsoft.CodeAnalysis.VisualBasic.LanguageVersion langVersion) Line 1464	C#
>	Roslyn.Services.UnitTests.dll!Microsoft.CodeAnalysis.UnitTests.CodeCleanup.RemoveUnnecessaryLineContinuationTests.DontRemoveLineContinuationInStringInterpolation3() Line 1377	C#

```

**Comment by davkean** [comment]
Unsure whom actually owns - i think it's one of our tests, but the assert in the compiler.

**Comment by Pilchie** [comment]
Moving to compilers based on the stack.

**Comment by AdamSpeight2008** [comment]
I know a little about `String.Format` and its parsing rules, as I implement an analyser to do some compile-time checks on them. I reimplement a new parser for `string.format`, to report multiple issues, not just the first exception.  *(Currently trying to find time to update it VS2015)*

**General String.Format Text Structure**
```
[  Format String                                                                ]
 [ Text ][EscOB][ ArgHole                                                       ]
                 [OB][ArgIndex    ] [ ArgAlign?          ] [ArgFormat?      ][CB]
                      [Digits][WS]   [,][WS][-?][Digits]   [ArgText][EscOB]
```


The escaping of braces within the **ArgFormat** Clause, `{ obj, -4 : x3}}}` => `0ff}}` is the same as itf you were escaping a brace within the **Text** part of a Format string. `{{{0,-4,x3}}}` The closing brace has to a be distinct singular brace. `}` to close the **ArgHole**

---------
@Pilchie  Do you have a line number? So I could have a look at it.

**Comment by AlekseyTs** [comment]
@davkean Please verify.
